AY2021S1-CS2103T-F11-4/tp
# Command history for CLI

## Description

Per title. Up button fetches previous valid command if available. Down button fetches next valid command if available.

Fixes #239 

## Testing

Manual testing
- Commands entered in order: `client view 1`, `list`
- Pressing up button correctly retrieves `list`. Pressing it again retrieves `client view 1`.
- Hit enter on `client view 1` to execute the command.
- Pressing up successfully retrieves `client view 1` from the history.

Automated:
- CommandHistory test to check history content is consistent.
- Upon entering a valid command, pressing up should get the latest command entered, not the second latest.

## Remarks

Only valid commands are added to the history, entering non valid commands do not add to history.
